src/components/createOnyxContext.tsx

```diff
@@ -33,12 +36,12 @@ type WithOnyxKey<TOnyxKey extends OnyxKeys> = <TNewOnyxKey extends string = TOny
 type CreateOnyxContext<TOnyxKey extends OnyxKeys> = [
     WithOnyxKey<TOnyxKey>,
     ComponentType<Omit<ProviderPropsWithOnyx<TOnyxKey>, TOnyxKey>>,
-    React.Context<OnyxKeyValue<TOnyxKey>>,
-    () => OnyxValues[TOnyxKey],
+    React.Context<OnyxContextType<TOnyxKey>>,
+    () => OnyxContextType<TOnyxKey> & (NonNullable<unknown> | undefined),
```

Comment:
Could you explain why we need the `& (NonNullable<unknown> | undefined)` part?

Reply:
ActuallyI am not sure why I needed to add that but after change to adjust `useOnyxContext` return type, the type for this hook looks like that 
<img width="1361" alt="image" src="https://github.com/Expensify/App/assets/25230417/30fc9f5d-f494-4ca9-bc71-e77cae8fb51a">
and this `& (NonNullable<unknown> | undefined)` is to replace ` & ({} | undefined)` becuase its produces lint error
@fabioh8010